crates/db_schema/src/impls/comment.rs

```diff
@@ -236,6 +236,20 @@ impl Comment {
     }
     Ok(())
   }
+
+  pub async fn update_comment_and_children(
+    pool: &mut DbPool<'_>,
+    comment_path: &Ltree,
+    form: &CommentUpdateForm,
+  ) -> LemmyResult<Vec<Self>> {
+    let conn = &mut get_conn(pool).await?;
+    diesel::update(comment::table)
+      .filter(comment::path.contained_by(comment_path))
```

Comment:
Nice, easy enough.

Reply:
I suppose youre right about the path, but maybe there is some edge case were not thinking of. I would feel safer with explicit post id. Maybe that will help with sql indexes too.

I would rather add params `locked` and `deleted` rather than take a form which allows lots of invalid states. Its also less verbose for callers that way.

Reply:
There's already an index on the path. If the query planner is not smart enough, the adding a post id filter might cause a scan that combines the post id index and the path index (for example, a bitmap scan), which is a lot worse.

This query is certainly correct. The case in which you need to be careful about it being correct is when changing how comment paths are stored (for example, removing the leading 0, or removing the trailing non-parent id number).

Reply:
Okay filters are fine like this. Still change it to take param `locked` instead of form to make call sites less verbose, and to avoid recursively changing other fields like content or others.